src/main/java/uk/gov/hmcts/reform/judicialbooking/domain/service/createbooking/CreateBookingOrchestrator.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.judicialbooking.domain.service.createbooking;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingEntity;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Booking;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.BookingRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.BookingResponse;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.JudicialUserProfile;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingAssignmentsRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.enums.Status;
+import uk.gov.hmcts.reform.judicialbooking.domain.service.common.OrgRoleMappingService;
+import uk.gov.hmcts.reform.judicialbooking.domain.service.common.ParseRequestService;
+import uk.gov.hmcts.reform.judicialbooking.domain.service.common.PersistenceService;
+import uk.gov.hmcts.reform.judicialbooking.domain.service.common.PrepareDataService;
+import uk.gov.hmcts.reform.judicialbooking.domain.service.common.RetrieveDataService;
+
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+
+@Slf4j
+@Service
+public class CreateBookingOrchestrator {
+
+    private final ParseRequestService parseRequestService;
+    private final PersistenceService persistenceService;
+    private final PrepareDataService prepareDataService;
+    private final RetrieveDataService retrieveDataService;
+    private final OrgRoleMappingService orgRoleMappingService;
+
+    BookingEntity bookingEntity;
```

Comment:
avoid creating any data member for this service class, this causes issues during concurrent execution as this class is singleton.